sdk/core/core-client/src/serviceClient.ts

```diff
@@ -29,8 +29,20 @@ export interface ServiceClientOptions extends CommonClientOptions {
   /**
    * If specified, this is the base URI that requests will be made against for this ServiceClient.
    * If it is not specified, then all OperationSpecs must contain a baseUrl property.
+   * @deprecated
    */
   baseUri?: string;
+  /**
+   * If specified, this is the endpoint that requests will be made against for this ServiceClient.
+   * If it is not specified, then all OperationSpecs must contain a baseUrl property.
+   * to encourage customer to use endpoint, we mark the baseUri as deprecated.
+   */
+  endpoint?: string;
```

Comment:
I think this also needs some runtime changes in ServiceClient.ts. When making the changes to consume endpoint please make sure to rename any internal uses of baseUri to endpoint. Also if both are present `endpoint` should win over baseUri